### ISSUE TITLE
Swapchain issues

### ISSUE THREAD

**Issue by ClonedPuppy** [GodotVR/godot-xr-template#23]
Hi, wanted to check this out to see how things are wired up for working in XR. Unfortunately I arbitrarily get a black viewport, more often that not. Looks like it's an issue of setting up the swapchain?

Here's what's in the log:

OpenXR: XR started (focused_state)
OpenXR: failed to release swapchain image! [ XR_ERROR_CALL_ORDER_INVALID ]
OpenXR: failed to release swapchain image! [ XR_ERROR_CALL_ORDER_INVALID ]
OpenXR: failed to acquire swapchain image [ XR_ERROR_CALL_ORDER_INVALID ]
OpenXR: No viewport was marked with use_xr, there is no rendered output!
OpenXR: failed to acquire swapchain image [ XR_ERROR_CALL_ORDER_INVALID ]
OpenXR: No viewport was marked with use_xr, there is no rendered output!
OpenXR: failed to acquire swapchain image [ XR_ERROR_CALL_ORDER_INVALID ]
OpenXR: No viewport was marked with use_xr, there is no rendered output!
---- cut. Continues until XR Ended, about 4000+ lines of above spam.

Running on a Quest Pro (57.0.0), cloned the xr-template as is without any modifications in the code. Builds successfully. Works as expected about 1 out of 10, or somewhere in that region.. hard to pin down.

Grateful for any feedback, perhaps just the state of XR on godot 4.1.1 Stable?

**Comment by BastiaanOlij** [comment]
Hmm, it's running fine on my Quest Pro but I've had issues running on Godot 4.2 but haven't been able to investigate it as the logging system is broken. May be the same issue.

**Comment by ClonedPuppy** [comment]
Yes, sorry for not being clear about that. Mobile renderer, that's correct. I wish I could give more details, but as I'm quite new on Godot, I'm still trying to wrap my head around things and how they work. In any case, I will keep an eye on the issue. I'm developing on .net so I'll be using desktop XR until there is support for Android export, so it's not of vital importance for me. Just thought I'd log it in case someone else has the same issue.

**Comment by BastiaanOlij** [comment]
Someone pointed out to me that it's also logging:

> OpenXR: No viewport was marked with use_xr, there is no rendered output!

Did you add the startup code that enable the XR mode on the main viewport? Without that it wont be sending images to OpenXR and that might be causing it to trip up, see also:
https://docs.godotengine.org/en/stable/tutorials/xr/setting_up_xr.html#setting-up-the-xr-scene

**Comment by ClonedPuppy** [comment]
No I did not.. I asumed(wrongly?) the template had all that covered, so I basically just installed the android export template and the openXR plugins. and pressed build/export. But as I mentioned, it does work sometime, especially when starting the template build from the editor. Seems a installed apk works less reliably.

**Comment by adammikulis** [comment]
I have been having that issue as well with 4.2, the template launched just fine on my Quest Pro before that but now it just launches a small black rectangle.

**Comment by ClonedPuppy** [comment]
I recently watched a VR prototyping video from Snoopek, and he kept running into the same issue that I do. It seems it's a known issue though(?), so hope they get it fixed before 4.2 stable.

**Comment by fisher2470** [comment]
I was running into the same issue, but adding the initialization script from the tutorial as a plugin in autoload, rather than attaching it to the root node of the scene resolves the error:

![image](https://github.com/GodotVR/godot-xr-template/assets/42970431/9460d880-fd33-4222-ab3c-a8d1a9b09162)

**Comment by BastiaanOlij** [comment]
Closing this as old. This has not been an issue for a long time asfar as I'm aware.

Interesting to see @fisher2470 seeing a difference between moving it to autoload. That suggests there was an issue in extending the script.

Note that Godot 4.x now requires explicit calls to overridden functions. This has been the cause of these types of issues as not calling `super()` in your overridden `_ready()` function often breaks code.
